init-distro-rid.sh

```diff
@@ -0,0 +1,195 @@
+            source "${rootfsDir}/etc/os-release"
+
+            # If we are doing a portable build. Then there are several cases
+            # where we must switch __PortableBuild=0 and use a non-portable
+            # distro rid. See the table above for a full list.
+            if (( ${isPortable} == 1 )); then
+                if [ "${ID}" = "rhel" ]; then
+                    # RHEL should have been caught by the /etc/redhat-release
+                    echo "Error, please verify that your install of RedHat includes"
+                    echo "/etc/redhat-release"
+                    exit 1
+                fi
+            else
+                # We have forced __PortableBuild=0. This is because -portablebuld
+                # has been passed as false.
+
+                if [ "${ID}" == "rhel" ]; then
+                    # remove the last version digit	
+                    VERSION_ID=${VERSION_ID%.*}
+                fi
+
+                nonPortableBuildID="${ID}.${VERSION_ID}-${buildArch}"
+            fi
+            
+        fi
+    fi
+
+    if [ "$buildOs" = "FreeBSD" ]; then
+        __freebsd_version=`sysctl -n kern.osrelease | cut -f1 -d'.'`
+        nonPortableBuildID="freebsd.$__freebsd_version-${buildArch}"
+    fi
+
+    if [ "${nonPortableBuildID}" != "" ]; then
+        export __DistroRid=${nonPortableBuildID}
+
+        # We are using a non-portable build rid. Force __PortableBuild to false.
+        export __PortableBuild=0
+    fi
+}
+
+
+# initDistroRidGlobal
+#
+# Input:
+#   os: (str)
+#   arch: (str)
+#   isPortable: (int)
+#   rootfsDir?: (nullable:string)
+#
+# Return:
+#   None
+#
+# Notes:
+#
+# The following out parameters are returned
+#
+#   __DistroRid
+#   __PortableBuild
+#
+initDistroRidGlobal()
+{
+    # __DistroRid must be set at the end of the function.
+    # Previously we would create a variable __HostDistroRid and/or __DistroRid.
+    #
+    # __HostDistroRid was used in the case of a non-portable build, it has been
+    # deprecated. Now only __DistroRid is supported. It will be used for both
+    # portable and non-portable rids and will be used in build-packages.sh
+
+    local buildOs=$1
+    local buildArch=$2
+    local isPortable=$3
+    local rootfsDir=$4
+
+    # Setup whether this is a crossbuild. We can find this out if rootfsDir
+    # is set. 
+    local isCrossBuild=0
+
+    if [ -z "${rootfsDir}" ]; then
+        isCrossBuild=0
+    else
+        # We may have a cross build. Check for the existance of the rootfsDir
+        if [ -e ${rootfsDir} ]; then
+            isCrossBuild=1
+        else
+            echo "Error rootfsDir has been passed, but the location is not valid."
+            exit 1
+        fi
+    fi
+
+    if (( ${isCrossBuild} == 1 )); then
+        initNonPortableDistroRid ${buildOs} ${buildArch} ${isPortable} ${isCrossBuild} ${rootfsDir}
+    else
+        initNonPortableDistroRid ${buildOs} ${buildArch} ${isPortable}
+    fi
```

Comment:
Could you just call this once, passing the rootfs dir (empty in the non-crossbuild case)?